infra/modules/vpc/main.tf

```diff
@@ -0,0 +1,45 @@
+locals {
+  project_name = var.project_name
+  vpc_name     = "vpc-${local.project_name}"
+  region       = var.region
+  cidr_prefix  = "172.30." # Value should be between "172.16." & "172.31."
+}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
```

Comment:
i had considered using third party modules when implementing the service module, but in the end decided against it for a couple of reasons:
- for security purposes, since infra is so sensitive, i'm not as comfortable depending on a third party module that isn't one of the "official" ones that are maintained by terraform, aws, etc
- the third party modules have a lot more functionality than we need and are therefore a bit harder to understand and modify if a project aims to do that, and a big goal of this template is to keep things simpler even at the cost of functionality

could we potentially just copy paste what we like from the terraform-aws-modules/vpc/aws module and create those resources directly?